wicketd/src/artifacts/extracted_artifacts.rs

```diff
@@ -143,12 +144,24 @@ impl ExtractedArtifacts {
                 })?,
         );
 
-        let file_size = io::copy(&mut reader, &mut writer)
-            .with_context(|| format!("failed writing to {output_path}"))
-            .map_err(|error| RepositoryError::CopyExtractedArtifact {
+        let mut stream = std::pin::pin!(stream);
+
+        let mut file_size = 0;
+
+        while let Some(res) = stream.next().await {
+            let chunk = res.map_err(|error| RepositoryError::ReadArtifact {
                 kind: artifact_hash_id.kind.clone(),
-                error,
-            })? as usize;
+                error: Box::new(error),
+            })?;
+            file_size += chunk.len();
+            writer
```

Comment:
Should this be a tokio BufWriter instead of std?

Reply:
It should, will fix.